### PR TITLE
Ensure assets-state.json is saved at end of backup even if there were no changes

### PR DESCRIPTION
**Pull request by jwodder**

### CODE REVIEW

**Review by yarikoptic — APPROVED** [review]
tests are still running but code looks good to me.  If tests are green, please merge and deploy and see if is in effect (how long update would take?)

File: tools/test_backups2datalad/test_zarr.py [hunk 1]
```diff
@@ -186,6 +186,7 @@ async def test_backup_zarr_delete_zarr(
     assert gitrepo.get_commit_subject("HEAD") == "[backups2datalad] 1 file deleted"
 
 
+@pytest.mark.xfail(reason="https://github.com/dandi/dandi-archive/issues/1245")
```
Comment (yarikoptic) [inline tools/test_backups2datalad/test_zarr.py:189]: backref: https://github.com/dandi/dandi-archive/issues/1245

Comment (yarikoptic) [reply tools/test_backups2datalad/test_zarr.py:189]: is it critical problem in that we should wait with this PR ?  or should we stop backingup for now?

Comment (jwodder) [reply tools/test_backups2datalad/test_zarr.py:189]: It only affects empty Zarrs, which shouldn't normally occur, but upon checking 000108, I see that there are 15 such Zarrs.  However, all of the Zarrs have non-`null` digests, so unless a new empty Zarr gets added sometimes soon, we should be fine.

Comment (djarecka) [reply tools/test_backups2datalad/test_zarr.py:189]: @jwodder - can you give me one example of the zarr, will check with people from mit

Comment (jwodder) [reply tools/test_backups2datalad/test_zarr.py:189]: @djarecka 

```
sub-MITU01/ses-20220322h10m15s10/micr/sub-MITU01_ses-20220322h10m15s10_sample-15_stain-NN_run-1_chunk-5_SPIM.ome.zarr
sub-MITU01/ses-20220310h18m01s48/micr/sub-MITU01_ses-20220310h18m01s48_sample-24_stain-NN_run-1_chunk-2_SPIM.ome.zarr
sub-MITU01/ses-20211222h12m30s05/micr/sub-MITU01_ses-20211222h12m30s05_sample-65_stain-NN_run-1_chunk-4_SPIM.ome.zarr
sub-MITU01/ses-20211214h20m22s55/micr/sub-MITU01_ses-20211214h20m22s55_sample-67_stain-NN_run-1_chunk-4_SPIM.ome.zarr
sub-MITU01/ses-20211222h12m30s05/micr/sub-MITU01_ses-20211222h12m30s05_sample-65_stain-YO_run-1_chunk-2_SPIM.ome.zarr
sub-MITU01/ses-20211214h20m22s55/micr/sub-MITU01_ses-20211214h20m22s55_sample-67_stain-NN_run-1_chunk-9_SPIM.ome.zarr
sub-MITU01/ses-20211222h12m30s05/micr/sub-MITU01_ses-20211222h12m30s05_sample-65_stain-YO_run-1_chunk-6_SPIM.ome.zarr
sub-MITU01/ses-20211222h12m30s05/micr/sub-MITU01_ses-20211222h12m30s05_sample-65_stain-YO_run-1_chunk-8_SPIM.ome.zarr
sub-MITU01/ses-20211214h20m22s55/micr/sub-MITU01_ses-20211214h20m22s55_sample-67_stain-YO_run-1_chunk-9_SPIM.ome.zarr
sub-MITU01/ses-20211222h12m30s05/micr/sub-MITU01_ses-20211222h12m30s05_sample-65_stain-NN_run-1_chunk-6_SPIM.ome.zarr
sub-MITU01/ses-20211214h20m22s55/micr/sub-MITU01_ses-20211214h20m22s55_sample-67_stain-YO_run-1_chunk-5_SPIM.ome.zarr
sub-MITU01/ses-20211222h12m30s05/micr/sub-MITU01_ses-20211222h12m30s05_sample-65_stain-NN_run-1_chunk-9_SPIM.ome.zarr
sub-MITU01/ses-20211214h20m22s55/micr/sub-MITU01_ses-20211214h20m22s55_sample-67_stain-NN_run-1_chunk-3_SPIM.ome.zarr
sub-MITU01/ses-20211222h12m30s05/micr/sub-MITU01_ses-20211222h12m30s05_sample-65_stain-YO_run-1_chunk-9_SPIM.ome.zarr
sub-MITU01/ses-20211214h20m22s55/micr/sub-MITU01_ses-20211214h20m22s55_sample-67_stain-YO_run-1_chunk-8_SPIM.ome.zarr
```

Comment (yarikoptic) [reply tools/test_backups2datalad/test_zarr.py:189]: thanks @jwodder, let's proceed then

Comment (djarecka) [reply tools/test_backups2datalad/test_zarr.py:189]: I don't see in my validation that these zarrs are empty...

The way how I read:
```
from dandi.dandiapi import DandiAPIClient
import zarr
import s3fs
fs = s3fs.S3FileSystem(anon=True)

dandiset = "000108"
api = DandiAPIClient("https://api.dandiarchive.org/api")
ds = api.get_dandiset(dandiset)

path = "sub-MITU01/ses-20211222h12m30s05/micr/sub-MITU01_ses-20211222h12m30s05_sample-65_stain-NN_run-1_chunk-4_SPIM.ome.zarr"
zarr_files = list(ds.get_assets_by_glob(path))
source = f"dandiarchive/zarr/{zarr_files[0].get_content_url(regex='s3').split('/')[-2]}/"

store_chunk = s3fs.S3Map(root=source, s3=fs, check=False)
root = zarr.group(store=store_chunk)
print(root[f"/6"][0,0,:,:,:].max())
```
Should we expect a different results?

Comment (jwodder) [reply tools/test_backups2datalad/test_zarr.py:189]: The sizes of the Zarrs as reported by the API are all zero.  If there are files on S3, then it probably means the Zarrs weren't ingested.

**Pull request merged**